keps/sig-testing/1746-stage-e2e-test-framework/README.md

```diff
@@ -0,0 +1,610 @@
+export any constants and variables defined in core Kubernetes that are required for e2e tests.
+
+### Managing Test Packages
+
+Decoupling the e2e test framework from other test packages will proceed in a
+similar manner.
+
+We will take a look at the utility functions surfaced by the e2e test framework
+and we will see how and why they are used in e2e tests.
+If we encounter utilities that are used sparingly in only a single test package
+then we will opt to move it to that test package.
+
+On the other hand, if we see a lot of dependencies and complexity introduced by
+requiring other test packages then we will work to migrate the essential parts
+within the e2e test framework.
+
+Ultimately, we want a library that is well encapsulated and has useful methods
+for maintainers to write e2e tests.
+
+
+### Notes
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+#### Kubemark
+
+One of the important e2e test components that we have been looking at is
+[Kubemark](https://github.com/kubernetes/kubernetes/tree/72cdc8c2112cf14daeb8b4f29cb524c3aba890b4/test/kubemark)
+which brings in a significant portion of dependencies from core Kubernetes.
+Some of the use of internal unpublished APIs is believed to be related to this
+and some other components built for scalability testing.
+
+Given that Kubemark is not part of the e2e test framework, the current plan is
+that no particular amount of work, other than the already mentioned, will be
+needed to decouple the e2e test framework from Kubemark.
+
+There is also the possibility that if Kubemark is the only reason why internal
+Kubernetes APIs (such as the Kubelet stat API) are needed that we move the codes
+that makes use of this to kubemark itself.
+
+
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the SIG or subproject.
+-->
+
+The work in this KEP involves one of the core components of e2e testing in the
+Kubernetes community.
+As with any kind of work, there is the risk that something will break.
+And since the area we are focusing on is such a core part of CI, we plan to use
+the vast set of e2e jobs and tests that already exist to monitor the health of our changes.
+
+In particular, we will focus on
+* https://testgrid.k8s.io/sig-release-master-blocking
+* https://testgrid.k8s.io/sig-release-master-informing
+
+These jobs are an integral part of ensuring that Kubernetes is working and in a good state to be used by people.
+Monitoring these jobs will help us make sure that our changes are correct.
+
+
+## Design Details
```

Comment:
this section probably should not be empty.

in what way do we expect to organize the packages such that this is "clean and reusable"?
how do we plan to accomplish the necessary decoupling? (import boss?)
where will the library be staged?

Reply:
Thank you for raising these questions.
Added more info in this section to try answer them.